### DISCUSSION TITLE
dialog.showSaveDialog seems never to be called

### DISCUSSION THREAD

**Discussion by ghost** [webdriverio-community/wdio-electron-service#879] [General]
Hello, I'm encontering an issue when mocking the Electron API dialog.showSaveDialog:

```
before(async () => {
        saveDialogMock = await browser.electron.mock(
            'dialog',
            'showSaveDialog',
            { canceled: false, filePath: 'output' }
        );
})
```

even if no dialog is shown after mocking and the mocked returned valued seems to be received from electron it seems that calling

```
        expect(saveDialogMock).toHaveBeenCalled();
```

throws always with 
```
Expected number of calls: >= 1
Received number of calls:    0
```
even it has been already called.

**Comment by goosewobbler** [comment]
@giovanni-bertoncelli How is this dialog function being triggered?  `showSaveDialog` is async, so you will need to ensure your test is awaiting the result of the triggering code, otherwise it may check the mock before it has been updated.

The service E2Es use [`showOpenDialog`](https://github.com/webdriverio-community/wdio-electron-service/blob/main/e2e/api.spec.ts#L9) for testing the mocking, I replaced that with `showSaveDialog` and they still pass.

**Comment by goosewobbler** [comment]
Converting to a discussion, we can always raise an issue from that if required
